jwst/master_background/expand_to_2d.py

```diff
@@ -329,8 +334,13 @@ def get_wavelengths(model):
             got_wavelength = False
             wl_array = None
 
+    # If no existing wavelength array, compute one
     if hasattr(model.meta, "wcs") and not got_wavelength:
-        wcs = model.meta.wcs
+        if hasattr(model.meta, "exposure") and model.meta.exposure.type == "NIS_SOSS":
+            transform = niriss.niriss_soss_set_input(model, order)
```

Comment:
It looks to me as if you could assign the value returned by `niriss_soss_set_input` directly to `wcs`, rather than first to `transform` and then to `wcs`.

Reply:
Yes, `niriss.niriss_soss_set_input` returns a WCS object/

Reply:
Done.